sklearn/preprocessing/_csr_polynomial_expansion.pyx

```diff
@@ -25,9 +51,29 @@ cdef inline INDEX_T _deg2_column(INDEX_T d, INDEX_T i, INDEX_T j,
     else:
         return d * i - (i**2 + i) / 2 + j
 
+def py_deg2_column(d, i, j, interaction_only):
+    """Compute the index of the column for a degree 2 expansion.
+
+    d is the dimensionality of the input data, i and j are the indices
+    for the columns involved in the expansion.
+
+    This function is defined in Python to use `PyLong`'s arbitrary
+    precision, and is called in Cython when C types' representability
+    range is too small.
+    """
+    if interaction_only:
+        return d * i - (i**2 + 3 * i) // 2 - 1 + j
```

Comment:
These are guaranteed to be divisible by 2, so are the // needed?

Reply:
This ensures the output is an `int` not a `float`